### PR TITLE
Icds extension case structure

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?191085

Nice edge case. Dumb mistake. 

Basically, because the closed case was a child, it was getting added as an "available" case and adding _all_ of the cases that were dependents of it to the sync. 

Wait for tests... 
@gcapalbo 
@czue

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -722,7 +722,7 @@ def _get_available_cases(self, relevant):
         incoming_extensions = _reverse_index_map(self.extension_index_tree.indices)
         available = {case for case in relevant
                      if case not in self.closed_cases
-                     and not self.extension_index_tree.indices.get(case) or self.index_tree.indices.get(case)}
+                     and (not self.extension_index_tree.indices.get(case) or self.index_tree.indices.get(case))}
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: ouch

**Comment by czue** [comment]
Is this also what's going on with http://manage.dimagi.com/default.asp?200797?

**Comment by proteusvacuum** [comment]
Could be. Looking into that now.

File: corehq/ex-submodules/casexml/apps/phone/models.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: :disappointed: 
I even remember wondering about this when I first wrote it.

**Pull request merged**